bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleRegistryImpl.java

```diff
@@ -91,75 +90,148 @@
 public class RuleRegistryImpl extends AbstractRegistry<Rule, String, RuleProvider>
         implements RuleRegistry, StatusInfoCallback {
 
+    private static final String SOURCE = RuleRegistryImpl.class.getSimpleName();
+    private static final Logger logger = LoggerFactory.getLogger(RuleRegistryImpl.class.getName());
+    private BundleContext bundleContext;
     private RuleEngine ruleEngine;
-    private Logger logger;
+    private StorageService storageService;
     private Storage<Boolean> disabledRulesStorage;
-    private TemplateManager templateManager;
+    private ModuleTypeRegistry moduleTypeRegistry;
+    private TemplateRegistry templateRegistry;
 
     /**
      * {@link Map} of template UIDs to rules where these templates participated.
      */
     private Map<String, Set<String>> mapTemplateToRules = new HashMap<String, Set<String>>();
-    @SuppressWarnings("rawtypes")
-    private ServiceTracker templateProviderTracker;
 
-    private static final String SOURCE = RuleRegistryImpl.class.getSimpleName();
+    public RuleRegistryImpl() {
+        super(RuleProvider.class);
+    }
 
-    @SuppressWarnings({ "rawtypes", "unchecked" })
-    public RuleRegistryImpl(RuleEngine ruleEngine, TemplateManager tManager, final BundleContext bc) {
-        super(null);
-        logger = LoggerFactory.getLogger(getClass());
-        this.ruleEngine = ruleEngine;
-        this.templateManager = tManager;
-        ruleEngine.setStatusInfoCallback(this);
-        templateProviderTracker = new ServiceTracker(bc, TemplateProvider.class.getName(),
-                new ServiceTrackerCustomizer() {
-
-                    @Override
-                    public Object addingService(ServiceReference reference) {
-                        TemplateProvider provider = (TemplateProvider) bc.getService(reference);
-                        templateUpdated(provider.getTemplates(null));
-                        return provider;
-                    }
+    /**
+     * Activates this component. Called from DS.
+     *
+     * @param componentContext this component context.
+     * @throws Exception
+     */
+    protected void activate(ComponentContext componentContext) throws Exception {
+        this.bundleContext = componentContext.getBundleContext();
+        this.ruleEngine = new RuleEngine(bundleContext, moduleTypeRegistry);
```

Comment:
If you consider RuleEngine as an internal implementation detail, you should not register it as a ManagedService, but rather have the RuleRegistry take a service configuration - this way, you would be able to remove the dependency on the bundle context and thus remove the OSGi references from the code.